sql/core/src/test/scala/org/apache/spark/sql/SparkSessionBuilderSuite.scala

```diff
@@ -152,4 +152,9 @@ class SparkSessionBuilderSuite extends SparkFunSuite with BeforeAndAfterEach {
       session.sparkContext.hadoopConfiguration.unset(mySpecialKey)
     }
   }
+
+  test("SPARK-27555: Spark SessionState.conf should load hive-site.xml ") {
+    val session = SparkSession.builder().master("local").getOrCreate()
+    assert(session.sessionState.conf.getConfString("hive.in.test") == "true")
```

Comment:
I'm not sure we want to do this. We load `hive-site.xml` into hadoop conf for hive compatibility. But `hive-site.xml` is not a native/official way to load Spark configs, so this conf should not appear in `SessionStage.conf`.

Please use the `spark-defaults.conf`.